apps/converter/src/selections.js

```diff
@@ -0,0 +1,440 @@
+import React  from "zarkit/react";
+import '@webcomponents/webcomponentsjs/webcomponents-loader.js';
+import '@material/mwc-button';
+import '@material/mwc-select';
+import '@material/mwc-list/mwc-list-item';
+import '@material/mwc-textfield';
+import Lengthfunc from "../Components/length";
+import AreaFunc from "../Components/Area";
+import DataFunc from "../Components/Datatransferrate";
+import DigitalStoragefunc from "../Components/DigitalStorage";
+import Energyfunc from "../Components/Energy ";
+import FrequencyFunc from "../Components/Frequency ";
+import MassFunc from "../Components/Mass ";
+import PressureFunc from "../Components/Pressure";
+import SpeedFunc from "../Components/Speed ";
+import TemperatureFunc from "../Components/Temperature ";
+import TimeFunc from "../Components/Time";
+class Selection extends React.Component {
+    constructor(props) {
+        super(props);
+
+    this.Lengthfunc = Lengthfunc.bind(this);
+    this.AreaFunc = AreaFunc.bind(this);
+    this.DataFunc = DataFunc.bind(this);
+    this.DigitalStoragefunc = DigitalStoragefunc.bind(this);
+    this.Energyfunc = Energyfunc.bind(this);
+    this.FrequencyFunc = FrequencyFunc.bind(this);
+    this.MassFunc = MassFunc.bind(this);
+    this.PressureFunc = PressureFunc.bind(this);
+    this.SpeedFunc = SpeedFunc.bind(this);
+    this.TemperatureFunc = TemperatureFunc.bind(this);
+    this.TimeFunc = TimeFunc.bind(this);
+
+    this.state = {query1 : ' ' , query2 : ' ', src:'',dest:'',formule:''};
+    }
+
+
+    p= (e) =>  {
+            var src,des,valsrc,valdest,result,madeSelection_1,madeSelection_2,pin;
+            var topselect=this.props.selectid;
+            if (e.target.id==='value') {
+                if (topselect==='Length'){ 
```

Comment:
these if statements could be replaced with an object that holds your condition and be more dynamic